sequencing/ATACSeq/jobSubmission/3_batchRunPeakCalling.sh

```diff
@@ -6,107 +6,158 @@
+	  sh ./ATACSeq/preprocessing/shiftAlignedReads.sh ${sample}
+  else
+    echo "Aligned bam file for ${sample} not found. Aligned reads can't be shifted."
+  fi
 fi
 
+## option PEAKS: peak calling is performed in two ways using MACS3: single-end and paired-end 
 if [ $# = 1 ] || [[ $2 =~ 'PEAKS' ]]
 then
-	echo Starting peak calling at:
-	date -u
+
 	module purge
-	module load MACS2/2.1.2.1-foss-2017b-Python-2.7.14
+	module load Python/3.9.6-GCCcore-11.2.0-bare
+  source /lustre/home/${USER}/pythonEnv/bin/activate
 	module load BEDTools
-	cd ${SCRIPTDIR}/
-	./ATACSeq/preprocessing/samplePeaks.sh ${sampleName}
-	
-	if [[ $? == 0 ]]
-		then date -u
-		echo "Peaks called"
-	fi
-	
-	
+ 
+  echo "Step 3.2 PEAKS started. Aligned reads will be used for peak calling using MACS3 PE and TA"
+  mapfile -t SAMPLES < ${METADIR}/samples.txt
+  sample=${SAMPLES[${SLURM_ARRAY_TASK_ID}]}
+  echo "Sample(s) specified by array number in command line is/are: " ${sample}
+  
+  mkdir -p ${PEAKDIR}/MACS/ShiftedTagAlign
+  mkdir -p ${PEAKDIR}/MACS/BAMPE
+  
+  if [[ -s ${ALIGNEDDIR}/${sample}.filt.nodup.bam ]]
+  then 
+    cd ${SCRIPTDIR}
+    sh ./ATACSeq/preprocessing/samplePeaks.sh ${sample}
+	else
+    echo "Aligned bam file for ${sample} not found. Peaks can't be called in aligned reads."
+  fi
 fi
 
+## option HMMRATAC: peak calling is performed using MACS3 HMMRATAC 
+if [ $# = 1 ] || [[ $2 =~ 'HMMRATAC' ]]
+then
+  
+  module purge
+  module load BEDTools
+  module load SAMtools
+  module load Python/3.9.6-GCCcore-11.2.0-bare
+  source /lustre/home/${USER}/pythonEnv/bin/activate
+  
+  echo "Step 3.2.1 HMMRATAC started. Aligned reads will be used for peak calling using MACS3 HMMRATAC."
+  mapfile -t SAMPLES < ${METADIR}/samples.txt
+  sample=${SAMPLES[${SLURM_ARRAY_TASK_ID}]}
+  echo "Sample(s) specified by array number in command line is/are: " ${sample}
+  
+  mkdir -p ${PEAKDIR}/HMMRATAC
+  
+  if [[ -s ${ALIGNEDDIR}/${sample}.filt.nodup.bam ]]
+  then 
+    cd ${SCRIPTDIR}
+    
+    sh ./ATACSeq/preprocessing/sampleHMMRatac.sh ${sample}
+  else
+    echo "Aligned bam file for ${sample} not found. Peaks can't be called in aligned reads."
+  fi
+fi
+
+## option FRIP: collate peak calling results statistics in a single csv file
 if [ $# = 1 ] || [[ $2 =~ 'FRIP' ]]
 then
-	echo Starting calculate FRIP:
-	date -u
+
 	module purge
 	module load BEDTools
 	module load SAMtools
-
+  
+  echo "Step 3.3 FRIP started. Peak calling statistics will be collated."
+  mapfile -t SAMPLES < ${METADIR}/samples.txt
+  
 	mkdir -p ${PEAKDIR}/QCOutput
 
 	cd ${SCRIPTDIR}/
-	sh ./ATACSeq/preprocessing/calcFrip.sh ${sampleName}
+  ## if peak calling done on both BAMPE and Shift Tag Align modes
+  #sh ./ATACSeq/preprocessing/calcFrip.sh ${sampleName}
 
-	if [[ $? == 0 ]]
-		then date -u
-		echo "FRIP calculated called"
-	fi
+  ## Outputs a single csv file with all peak calling results for all samples
+  sh ./ATACSeq/preprocessing/collateCalcFrip.sh ${SAMPLES[@]}
```

Comment:
I presume this will come in a later part of this PR, as I don't see this script anywhere